### ISSUE TITLE
Whispering in Character Makes Anonymous Very Confused

### ISSUE THREAD

**Issue by lucky-trill** [reonZ/anonymous#2]
If you whisper as a specific actor whose name is hidden with Anonymous, it will replace every single letter in the message.

DM view:
![image](https://user-images.githubusercontent.com/109553278/205990670-f28d17f6-63fe-4b0f-97f1-315ab421c15e.png)

Even more hilarious player view:
![image](https://user-images.githubusercontent.com/109553278/205990964-ec2495b0-61ac-4eff-9501-c75b138c30a1.png)

This is fairly niche (as core Foundry makes whispering as an actor fairly difficult), but it would be nice if it didn't do that.

**Comment by reonZ** [comment]
Wow, i have no idea why it would do that, i can't even wrap my mind on it.

I will investigate.

**Comment by reonZ** [comment]
You probably have another module or your system interfering in some way with chat message rendering.

I can't even impersonate an actor when whispering someone as the GM, i have to do it programmatically.

I did some testing sending whispers via code and the result was completely normal.

GM perspective
![Foundry_Virtual_Tabletop_biRtmEm2s9](https://user-images.githubusercontent.com/651462/206018715-4129eff1-33f5-4949-9642-31b116f73a68.jpg)

Player perspective:
![firefox_pki7CuRVHX](https://user-images.githubusercontent.com/651462/206018838-c7ea23f3-168c-4663-951e-1468d0b5f87a.jpg)

**Comment by lucky-trill** [comment]
My screencap above was in a fresh test world with no other modules activated. To double-check potential sources of conflict, I went ahead and did a totally fresh install of the Electron app. Tested in 5e with other else active - same as above. Tried Starfinder too to see if was system-specific - same as above:

![image](https://user-images.githubusercontent.com/109553278/206059981-e638bd3a-ac79-401a-a2a1-4c4e847f5e3c.png)

Whispering as a character does have to be done with macros or otherwise programmatically, so it could be an issue with the script, but what I use for testing purposes is pretty basic:

```ChatMessage.create({
                user: "[INSERT YOUR ID HERE]",
                speaker: {
                    actor: "[INSERT ACTOR ID HERE]"
                },
                content: "Test message here.",
                whisper: "[INSERT TARGET USER ID HERE]"
            });```

It also could be Foundry version specific. Both my main instance and the re-install are using Foundry 10.291.

**Comment by lucky-trill** [comment]
Apologies for the accidental close. Just fixing that.

**Comment by reonZ** [comment]
I dunno what to tell you, i did the test and nothing weird happened.
The screenshots i posted earlier were taken from a test on the `dnd5e` system, and the ones following are done with the `pf2e` system.

GM perspective:
![Foundry_Virtual_Tabletop_ob0cQwpbM4](https://user-images.githubusercontent.com/651462/206073465-c126e6cb-95cf-4973-a8b6-f38fe58b5168.jpg)

Player perspective:
![firefox_momM7Ryvpy](https://user-images.githubusercontent.com/651462/206073575-ec427d40-63b6-41a1-8d9e-eafbf53de7e7.jpg)

As you can see, everything is fine.

**Comment by reonZ** [comment]
Nevermind, i found the problem, it is because you are not passing anything but the actor id in the speaker, i will make a fix for such situation.

Just an fyi, you should use `ChatMessage.getSpeaker` to provide a chat message speaker. But i should still take that fringe scenario into account.

**Comment by reonZ** [comment]
This has been fixed with version `1.0.5`.

Thanks for the help.
